packages/clerk-js/src/ui/components/OrganizationProfile/ActionConfirmationPage.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+
+import { useWizard, Wizard } from '../../common';
+import { useCoreOrganization, useCoreUser } from '../../contexts';
+import { LocalizationKey, localizationKeys, Text } from '../../customizables';
+import { Form, useCardState, withCardStateProvider } from '../../elements';
+import { useNavigate } from '../../hooks';
+import { handleError } from '../../utils';
+import { FormButtonContainer } from '../UserProfile/FormButtons';
+import { SuccessPage } from '../UserProfile/SuccessPage';
+import { ContentPage } from './OrganizationContentPage';
+
+export const LeaveOrganizationPage = () => {
+  const { organization, membership } = useCoreOrganization();
+  const user = useCoreUser();
+
+  if (!organization || !membership) {
+    return null;
+  }
+
+  const leave = () => {
+    return organization.removeMember(user.id);
+  };
+
+  return (
+    <ActionConfirmationPage
+      title={localizationKeys('organizationProfile.profilePage.leave.title')}
+      messageLine1={localizationKeys('organizationProfile.profilePage.leave.messageLine1')}
+      messageLine2={localizationKeys('organizationProfile.profilePage.leave.messageLine2')}
+      successMessage={localizationKeys('organizationProfile.profilePage.leave.successMessage')}
+      onConfirmation={() => Promise.resolve(leave())}
```

Comment:
Why do we need the `Promise.resolve`? I'd expect something like this:

```suggestion
      onConfirmation={leave}
```

Reply:
You're right. Forgot to refactor that. Fixed!